### PR TITLE
feat(SpokePoolVerifier): Support calling depositV3 instead of deposit and deploy new contracts

### PR DESCRIPTION
**Pull request by nicholaspai**
We are moving to deprecate all legacy deposit() calls

### CODE REVIEW

**Review by dohaki — APPROVED** [review]

File: contracts/SpokePoolVerifier.sol [hunk 1]
```diff
@@ -23,41 +23,51 @@ contract SpokePoolVerifier {
...
+        address inputToken,
+        uint256 inputAmount,
+        uint256 outputAmount,
         uint256 destinationChainId,
-        int64 relayerFeePct,
         uint32 quoteTimestamp,
         bytes memory message,
-        uint256 maxCount
+        address exclusiveRelayer,
+        uint32 exclusivityDeadline,
```
Comment (bmzig) [inline contracts/SpokePoolVerifier.sol:50]: Should we reorder the arguments so the calldata is smaller? Might also be nice to use the same structure as `depositV3` just for organization purposes.

**Review by bmzig — APPROVED** [review]

**Review by pxrl — APPROVED** [review]
lgtm!

**Review by mrice32 — APPROVED** [review]
LGTM! This should be tagged as pre-audit.

**Comment by nicholaspai** [comment]
> LGTM! This should be tagged as pre-audit.

How do you suggest we do this?

**Pull request merged**